config-service-impl/src/main/java/org/hypertrace/config/service/store/DocumentConfigStore.java

```diff
@@ -208,30 +212,89 @@ public Map<ConfigResourceContext, ContextSpecificConfig> getContextConfigs(
   }
 
   @Override
-  public List<ContextSpecificConfig> getAllConfigs(ConfigResource configResource)
+  public List<ContextSpecificConfig> getAllConfigs(
+      ConfigResource configResource,
+      org.hypertrace.config.service.v1.Filter filter,
+      org.hypertrace.config.service.v1.Pagination pagination,
+      List<SortBy> sortByList)
       throws IOException {
-    Query query =
-        Query.builder()
-            .addSort(IdentifierExpression.of(VERSION_FIELD_NAME), SortOrder.DESC)
-            .setFilter(getConfigResourceFilterTypeExpression(configResource))
-            .build();
-    List<ContextSpecificConfig> contextSpecificConfigList = new ArrayList<>();
+
+    Query query = buildQuery(configResource, filter, pagination, sortByList);
+    List<ContextSpecificConfig> configList = new ArrayList<>();
     Set<String> seenContexts = new HashSet<>();
+
     try (CloseableIterator<Document> documentIterator =
         collection.query(query, QueryOptions.DEFAULT_QUERY_OPTIONS)) {
       while (documentIterator.hasNext()) {
-        String documentString = documentIterator.next().toJson();
-        ConfigDocument configDocument = ConfigDocument.fromJson(documentString);
-        String context = configDocument.getContext();
-        if (seenContexts.add(context)) {
-          convertToContextSpecificConfig(configDocument).ifPresent(contextSpecificConfigList::add);
-        }
+        processDocument(documentIterator.next(), seenContexts, configList);
       }
     }
-    Collections.sort(
-        contextSpecificConfigList,
+
+    configList.sort(
```

Comment:
We should only sort by creation timestamp if there's no requested sort - this is basically the default case and is overwriting the sort applied by the query.

Reply:
done